numba_cuda/numba/cuda/cudadrv/nvrtc.py

```diff
@@ -1,292 +1,52 @@
 # SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 # SPDX-License-Identifier: BSD-2-Clause
 
-from ctypes import byref, c_char, c_char_p, c_int, c_size_t, c_void_p, POINTER
-from enum import IntEnum
 from numba.cuda.cudadrv.error import (
     CCSupportError,
     NvrtcError,
-    NvrtcBuiltinOperationFailure,
-    NvrtcCompilationError,
-    NvrtcSupportError,
 )
 from numba.cuda import config
 from numba.cuda.cuda_paths import get_cuda_paths
 from numba.cuda.utils import _readenv
 
-import functools
 import os
-import threading
 import warnings
+import functools
 
 NVRTC_EXTRA_SEARCH_PATHS = _readenv(
     "NUMBA_CUDA_NVRTC_EXTRA_SEARCH_PATHS", str, ""
 ) or getattr(config, "CUDA_NVRTC_EXTRA_SEARCH_PATHS", "")
 if not hasattr(config, "CUDA_NVRTC_EXTRA_SEARCH_PATHS"):
     config.CUDA_NVRTC_EXTRA_SEARCH_PATHS = NVRTC_EXTRA_SEARCH_PATHS
 
-# Opaque handle for compilation unit
-nvrtc_program = c_void_p
-
-# Result code
-nvrtc_result = c_int
-
-if config.CUDA_USE_NVIDIA_BINDING:
-    from cuda.bindings import nvrtc as bindings_nvrtc
-    from cuda.core.experimental import Program, ProgramOptions
-
-
-class NvrtcResult(IntEnum):
-    NVRTC_SUCCESS = 0
-    NVRTC_ERROR_OUT_OF_MEMORY = 1
-    NVRTC_ERROR_PROGRAM_CREATION_FAILURE = 2
-    NVRTC_ERROR_INVALID_INPUT = 3
-    NVRTC_ERROR_INVALID_PROGRAM = 4
-    NVRTC_ERROR_INVALID_OPTION = 5
-    NVRTC_ERROR_COMPILATION = 6
-    NVRTC_ERROR_BUILTIN_OPERATION_FAILURE = 7
-    NVRTC_ERROR_NO_NAME_EXPRESSIONS_AFTER_COMPILATION = 8
-    NVRTC_ERROR_NO_LOWERED_NAMES_BEFORE_COMPILATION = 9
-    NVRTC_ERROR_NAME_EXPRESSION_NOT_VALID = 10
-    NVRTC_ERROR_INTERNAL_ERROR = 11
-
-
-_nvrtc_lock = threading.Lock()
-
-
-class NvrtcProgram:
-    """
-    A class for managing the lifetime of nvrtcProgram instances. Instances of
-    the class own an nvrtcProgram; when an instance is deleted, the underlying
-    nvrtcProgram is destroyed using the appropriate NVRTC API.
-    """
-
-    def __init__(self, nvrtc, handle):
-        self._nvrtc = nvrtc
-        self._handle = handle
-
-    @property
-    def handle(self):
-        return self._handle
+try:
+    # Prefer cuda.core for compilation
+    from cuda.core.experimental import Program, ProgramOptions  # type: ignore
 
-    def __del__(self):
-        if self._handle:
-            self._nvrtc.destroy_program(self)
+    _HAVE_CORE = True
+except Exception:  # pragma: no cover - environment dependent
+    _HAVE_CORE = False
 
+try:
+    # Use bindings for version / supported archs and as fallback for gaps
+    from cuda.bindings import nvrtc as bindings_nvrtc  # type: ignore
 
-class NVRTC:
-    """
-    Provides a Pythonic interface to the NVRTC APIs, abstracting away the C API
-    calls.
+    _HAVE_BINDINGS = True
+except Exception:  # pragma: no cover - environment dependent
+    bindings_nvrtc = None
+    _HAVE_BINDINGS = False
```

Comment:
This seems unnecessary, given that `cuda.bindings` was imported before along with `cuda.core` only if `USE_NV_BINDING` was `True`. Since that's always `True`, why are we adding additional import guards?